### PR TITLE
test(topup): implement test cases for the topup feature

### PR DESCRIPTION
**Pull request by ctrlc03**
# Description

Implement test cases across the codebase for the topup feature.

## Related issue(s)

Re: #923 

## Confirmation

- [x] I have read and understand MACI's [contributor guidelines](https://github.com/privacy-scaling-explorations/maci/blob/dev/CONTRIBUTING.md) and [code of conduct](https://github.com/privacy-scaling-explorations/maci/blob/dev/CODE_OF_CONDUCT.md).
- [x] I have read and understand MACI's [GitHub processes](https://github.com/privacy-scaling-explorations/maci/discussions/847).
- [x] I have read and understand MACI's [testing guide](https://maci.pse.dev/docs/testing).

### CODE REVIEW

File: cli/tests/e2e.test.ts [hunk 1]
```diff
@@ -1236,4 +1238,83 @@ describe("e2e tests", function () {
...
     });
   });
+
+  describe("topup message", () => {
+    const user = new Keypair();
+    const tokenAmount = 100;
+    let stateIndex: number | undefined;
+    after(async () => {
+      cleanVanilla();
+    });
```
Comment (0xmad) [inline cli/tests/e2e.test.ts:1248]:
```suggestion
    let stateIndex: number | undefined;

    after(() => {
      cleanVanilla();
    });
```

File: contracts/tests/Poll.test.ts [hunk 2]
```diff
@@ -126,6 +128,41 @@ describe("Poll", () => {
     });
   });
 
+  describe("topup", () => {
+    let voter: Signer;
+    before(async () => {
```
Comment (0xmad) [inline contracts/tests/Poll.test.ts:133]: Let's add new lines between hooks and tests

File: cli/test.sh
Comment (0xmad) [inline cli/test.sh:1]: Is this file useful for manual checks or is it using somewhere else?

**Review by 0xmad — CHANGES_REQUESTED** [review]
@ctrlc03 thanks, just few code style comments.

File: cli/test.sh
Comment (ctrlc03) [reply cli/test.sh:1]: oh did not meant to push this, it's a script I run sometimes to test e2e

Comment (0xmad) [reply cli/test.sh:1]: I think we can leave it. It will be useful for manual smoke tests.

File: cli/tests/e2e.test.ts [hunk 3]
```diff
@@ -1236,4 +1238,84 @@ describe("e2e tests", function () {
       await verify("0", testTallyFilePath);
     });
   });
+
+  describe("topup message", () => {
+    const user = new Keypair();
+    const tokenAmount = 100;
+    let stateIndex: number | undefined;
+
+    after(async () => {
```
Comment (0xmad) [inline cli/tests/e2e.test.ts:1247]: No need in async here.
```suggestion
    after(() => {
```

**Review by 0xmad — APPROVED** [review]
@ctrlc03 thanks!

**Pull request merged**